src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -939,6 +939,7 @@ default void setUtf8String(long offset, String str) {
      *
      * @param buffer the buffer instance to be turned into a new memory segment.
      * @return a memory segment, derived from the given buffer instance.
+     * @throws IllegalArgumentException if the provided {@code buffer} is a StringCharBuffer.
```

Comment:
Note that StringCharBuffer is not a public API type. It is something we create with `CharBuffer,wrap(CharSequence)` so the doc should reflect that.